metrics/metrics.go

```diff
@@ -63,6 +99,9 @@ func Middleware(registry prometheus.Registerer) gin.HandlerFunc {
 		count.Inc()
 		defer count.Dec()
 
+		metricContext := context.WithValue(c.Request.Context(), providerMetricsKey, outboundCallDuration)
```

Comment:
Adding this to the context so that we can observe the metric where it is called. The other option I could come up with was to add different call durations to the context and check them at the end.

Reply:
I'm not sure about setting up this metric like this. The indirection makes it harder to trace. The metrics middleware is also not the right place to create/register this metric since it's shared between the server and connector. Instead it should probably be registered in `server/metrics.go`

Reply:
Added it directly in-line with the `syncIdentityInfo` function